### PR TITLE
feat(plugin-vue): add `customElement` option to compiler

### PR DESCRIPTION
**Pull request by baiwusanyu-c**
### Description

According to the filter result of vite-plugin-vue, the compiler like vue passes the customElement option parameter

### Additional context

~~wait upstream https://github.com/vuejs/core/pull/8989~~

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md#pull-request-guidelines) and follow the [Commit Convention](https://github.com/vitejs/vite-plugin-vue/blob/main/.github/commit-convention.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/plugin-vue/src/script.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export function resolveScript(
   descriptor: SFCDescriptor,
   options: ResolvedOptions,
   ssr: boolean,
+  asCustomElement: boolean,
```
Comment (sxzz) [inline packages/plugin-vue/src/script.ts:51]:
```suggestion
  customElement: boolean,
```

maybe shorter?

File: packages/plugin-vue/src/script.ts [hunk 2]
```diff
@@ -72,6 +73,7 @@ export function resolveScript(
     genDefaultAs: canInlineMain(descriptor, options)
       ? scriptIdentifier
       : undefined,
+    customElement: asCustomElement,
```
[the remark this answers is not in the document]
Comment (sxzz) [reply packages/plugin-vue/src/script.ts]: Same for rest part.

File: packages/plugin-vue/src/main.ts [hunk 3]
```diff
@@ -61,6 +61,7 @@ export async function transformMain(
     options,
     pluginContext,
     ssr,
+    asCustomElement,
```
Comment (sxzz) [inline packages/plugin-vue/src/main.ts:64]: same here

File: packages/plugin-vue/src/index.ts [hunk 4]
```diff
@@ -10,7 +10,6 @@ import type {
 } from 'vue/compiler-sfc'
 import type * as _compiler from 'vue/compiler-sfc'
 /* eslint-enable import/no-duplicates */
-import { computed, shallowRef } from 'vue'
```
[the remark this answers is not in the document]
Comment (baiwusanyu-c) [reply packages/plugin-vue/src/index.ts]: I accidentally deleted it, I will restore it now

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
Looks like CI failed.

**Pull request merged**